pubsub/natspubsub/nats.go

```diff
@@ -160,24 +159,30 @@ func (*topic) ErrorCode(err error) gcerrors.ErrorCode {
 }
 
 type subscription struct {
-	nc   *nats.Conn
-	nsub *nats.Subscription
-	err  error
+	nc      *nats.Conn
+	nsub    *nats.Subscription
+	ackFunc func()
+	err     error
 }
 
 // CreateSubscription returns a *pubsub.Subscription representing a NATS subscription.
 // TODO(dlc) - Options for queue groups?
-func CreateSubscription(nc *nats.Conn, subscriptionName string) *pubsub.Subscription {
-	return pubsub.NewSubscription(createSubscription(nc, subscriptionName), nil)
+func CreateSubscription(nc *nats.Conn, subscriptionName string, ackFunc func()) *pubsub.Subscription {
+	return pubsub.NewSubscription(createSubscription(nc, subscriptionName, ackFunc), nil)
 }
 
-func createSubscription(nc *nats.Conn, subscriptionName string) driver.Subscription {
+func createSubscription(nc *nats.Conn, subscriptionName string, ackFunc func()) driver.Subscription {
```

Comment:
Check that ackFunc is non-nil. If it is nil, make an error and stick it into the subscription.

Reply:
It seems that that error wasn't being used anywhere. I've modified the constructors to return errors explicitly so we can validate arguments.